models/lobby/server.go

```diff
@@ -0,0 +1,255 @@
+
+	// check if all players in server are in lobby
+	s.Players = s.Rcon.GetPlayers()
+	for i := range s.Players {
+		// check if player is not in lobby but is in server
+		// ignores BOT (SourceTV)
+		if s.Players[i].SteamID != "BOT" && s.IsPlayerInLobby(s.Players[i].SteamID) == false {
+			fmt.Println("[Server.Verify]: Kicking player not allowed -> Username [" +
+				s.Players[i].Username + "] SteamID [" + s.Players[i].SteamID + "]")
+
+			kickErr := s.Rcon.KickPlayer(s.Players[i], "[TeamPlay.TF]: You're not in this lobby...")
+
+			if kickErr != nil {
+				log.Fatal(kickErr)
+			}
+		}
+	}
+
+}
+
+// check if the given steamId is in the server
+func (s *Server) IsPlayerInServer(steamId string) bool {
+	for i := range s.Players {
+		if steamId == s.Players[i].SteamID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// check if the given steamId is in the allowedPlayers list
+func (s *Server) IsPlayerInLobby(steamId string) bool {
+	// SourceTV
+	if steamId == "BOT" {
+		return false
+	}
+
+	for i := range s.AllowedPlayers {
+		if steamId == s.AllowedPlayers[i].SteamID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// TODO: get end event from logs
+// `World triggered "Game_Over"`
+func (s *Server) End() {
+	fmt.Println("[Server.End]: Ending server -> [" + s.Address + "] from lobby [" + strconv.Itoa(s.LobbyId) + "]")
+	// TODO: upload logs
+
+	s.Rcon.Close()
+	s.Ticker.Close()
+}
+
+func (s *Server) ExecConfig(config *ServerConfig) error {
+	fmt.Println("[Server.ExecConfig]: Running config!")
+	configErr := s.Rcon.ExecConfig(config.Data)
+
+	if configErr != nil {
+		fmt.Println("[Server.ExecConfig]: Error while trying to run config!")
+
+		return configErr
+	}
+
+	return nil
+}
+
+func (s *Server) KickAll() error {
+	fmt.Println("[Server.KickAll]: Kicking players...")
+	s.Players = s.Rcon.GetPlayers()
+
+	for i := range s.Players {
+		kickErr := s.Rcon.KickPlayer(s.Players[i], "[TeamPlay.TF]: Setting up lobby...")
+
+		if kickErr != nil {
+			return kickErr
+		}
+	}
+
+	return nil
+}
+
+func (s *Server) ChangeMap(newMap string) error {
+	fmt.Println("[Server.ChangeMap]: Changing [" + s.Address + "]'s map to [" + newMap + "]...")
+	_, queryErr := s.Rcon.Query("changelevel " + newMap)
+
+	return queryErr
+}
+
+func (s *Server) ChangePassword(newPassword string) error {
+	// reset password if length is 0
+	// command would be: sv_password ""
+	if newPassword == "" {
+		newPassword = `""`
+	}
+
+	fmt.Println("[Server.ChangePassword]: Changing [" + strconv.Itoa(s.LobbyId) + " @ " + s.Address + "]'s password...")
+	_, queryErr := s.Rcon.Query("sv_password " + newPassword)
+
+	if queryErr == nil {
+		fmt.Println("[Server.ChangePassword]: Changed [" + strconv.Itoa(s.LobbyId) + " @ " + s.Address + "]'s password!")
+	}
+
+	return queryErr
+}
+
+func (s *Server) AllowPlayer(steamId string) {
+	s.AllowedPlayers = append(s.AllowedPlayers, TF2RconWrapper.Player{SteamID: steamId})
```

Comment:
again, keep in mind steamid types. Wow, this is confusing. Maybe we should make them different type aliases?